kingpin/actors/aws/ecs.py

```diff
@@ -406,7 +420,369 @@ def _execute(self):
         if self.option('wait'):
             yield self._wait_for_tasks(tasks)
         else:
-            self.log.info('Not waiting for tasks to stop')
+            self.log.info('Not waiting for tasks to complete')
+
+
+# http://boto3.readthedocs.io/en/latest/reference/services/ecs.html
+SERVICE_DEFINITION_SCHEMA = {
+    'type': 'object',
+    'additionalProperties': False,
+    'properties': {
+        'loadBalancers': {
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {
+                    'loadBalancerName': {'type': 'string'},
+                    'containerName': {'type': 'string'},
+                    'containerPort': {'type': 'number'},
+                }
+            }
+        },
+        'role': {'type': 'string'},
+        'deploymentConfiguration': {
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'maximumPercent': {'type': 'number'},
+                'minimumHealthyPercent': {'type': 'number'},
+            }
+        }
+    }
+}
+
+
+class DeployService(ECSBaseActor):
+    """Register and run a service on ECS.
+
+    This actor will loop indefinitely until the task is complete.
+
+    If the service already exists, it is upgraded.
+
+    **Options**
+
+    :region:
+      AWS region (or zone) name, such as us-west-2 or eu-west-1.
+
+    :cluster:
+      ECS cluster in the region to launch the task into.
+
+    :task_definition:
+      String of path to the Task Definition file template.
+      Must be a local file path.
+      Tokens to be interpolated must be of the form %VAR%.
+
+    :service_definition:
+      String of path to the Service Definition file template.
+      Must be a local file path.
+      Tokens to be interpolated must be of the form %VAR%.
+      Implicit fields - do not include these:
+      'serviceName', 'taskDefinition', 'desiredCount', 'clientToken'
+      Used for: 'loadBalancers', 'role', 'deploymentConfiguration'
+      Default: None.
+
+    :service_name:
+      Service name to use. If not specified, this will use
+      the Task Definition's family.
+      Default: None.
+
+    :tokens:
+      A dictionary of key/value pairs used to fill in the tokens for the
+      Task and Service Definition template. Default: {}.
+
+    :count:
+      How many instances of the service to deploy.
+      Default: 1.
+
+    :wait:
+      Whether to wait for the services to deploy.
+      Default: True.
```

Comment:
We have a global actor timeout value. I'd keep this as boolean and encourage using the `timeout` param : http://kingpin.readthedocs.io/en/stable/basicuse.html?highlight=timeout#schema-description